### PR TITLE
SSL support

### PR DESCRIPTION
**Pull request by allanbartley**
Add support for SSL connections.
Requires esp8266/Arduino 2.4.0, which is not yet released (platform = espressif8266_stage in PlatformIO).
Requires SSL to be explicitly enabled via build_flags (build_flags = -DASYNC_TCP_SSL_ENABLED=1 in PlatformIO).
Fingerprint checking is supported.

### CODE REVIEW

File: docs/2.-API-reference.md [hunk 1]
```diff
@@ -62,6 +62,18 @@ Set the server.
 * **`host`**: Host of the server
 * **`port`**: Port of the server
 
+#### AsyncMqttClient& setSecure(bool `secure`)
+
+Whether or not to use SSL. Defaults to `false`.
+
+* **`secure`**: SSL wanted or not.
+
+#### AsyncMqttClient& setServerFingerprint(const uint8_t\* `fingerprint`)
```
Comment (marvinroger) [inline docs/2.-API-reference.md]: As you say, this method *adds* a fingerprint. Can you please rename this method to `addServerFingerprint`?

File: src/AsyncMqttClient.cpp [hunk 2]
```diff
@@ -141,6 +158,25 @@ void AsyncMqttClient::_clear() {
...
+    SSL* clientSsl = _client.getSSL();
+    const uint8_t* fingerprint;
+    for (std::vector<std::array<uint8_t, SHA1_SIZE>>::iterator it = _secureServerFingerprints.begin(); it != _secureServerFingerprints.end(); ++it) {
+      fingerprint = it->data();
+      if (ssl_match_fingerprint(clientSsl, fingerprint) == SSL_OK) {
+        sslFoundFingerprint = true;
+      }
+    }
+    if (!sslFoundFingerprint) {
+      _client.close();
```
Comment (marvinroger) [inline src/AsyncMqttClient.cpp]: Please replace this line with `_client.close(true)`. There's a reconnection problem without it.

File: src/AsyncMqttClient.cpp [hunk 3]
```diff
@@ -92,6 +95,20 @@ AsyncMqttClient& AsyncMqttClient::setServer(const char* host, uint16_t port) {
   return *this;
 }
 
+#if ASYNC_TCP_SSL_ENABLED
+AsyncMqttClient& AsyncMqttClient::setSecure(bool secure) {
```
Comment (marvinroger) [inline src/AsyncMqttClient.cpp]: Can you add this to the `keywords.txt` file please?

File: docs/2.-API-reference.md (hunk 1 above)
Comment (marvinroger) [inline docs/2.-API-reference.md]: Also, please add this to the `keywords.txt`.

File: src/AsyncMqttClient.cpp (hunk 2 above)
Comment (allanbartley) [inline src/AsyncMqttClient.cpp]: Update made.

File: docs/2.-API-reference.md (hunk 1 above)
Comment (allanbartley) [inline docs/2.-API-reference.md]: Function renamed, added to keywords.txt.

File: src/AsyncMqttClient.cpp (hunk 3 above)
Comment (allanbartley) [inline src/AsyncMqttClient.cpp]: Function added to keywords.txt.

**Comment by marvinroger** [comment]
Thanks a lot! 👍

**Pull request merged**